scripts/transformations.py

```diff
@@ -20,130 +20,137 @@
+                                [0, 0, 1]])
+
+            # todo check multiplication order
+            rotation_matrix = np.matmul(rot, rotation_matrix)
+
+        return rotation_matrix
+
+    # todo: possibly remove? Decision to be made by @ACBuynak
+    # previously convertPath2FixedFrame
+    @staticmethod
+    def convertPath2FixedFrame(path_body, frame_body, frame_fixed=np.identity((4))):
+        """
+        Function for mapping a Path (type: List) onto a Fixed Frame (np. Homogenous Matrix)
+
+        @param path_body: Python List of Transformation Matrices for each point along a path
+        @param frame_body: Transformation matrix for the body
+        @param frame_fixed: Defaults to Identity Matrix (or robot's fixed frame)
+        """
+
+        ## Find Arb Frame Defined By Fixed Frame
+        frame_b2f = np.matmul(frame_body, frame_fixed)  #Pre-Multiply bc wrt fixed frame
+
+        ## Convert Path Frames to be defined by Fixed Frame
+        path_fixed = []
+        for point in path_body:
+            path_fixed.append(np.matmul(frame_b2f, point))
+
+        return path_fixed
+
+    @staticmethod
+    def convert_transformations_to_poses(path_transformations):
+        # type: (List[np.ndarray]) -> List[List[float]]
+        """
+        Converts a path of homogenous transformations to a path of
+        poses (x, y, z, qx, qy, qz, qw)
+
+        Function based upon math in...
+        https://en.wikipedia.org/wiki/Rotation_matrix#Quaternion
+
+        @param path_transformations: List of Homogeneous Transformations
+
+        @return: List of Robot Poses in the form [x, y, z, qx, qy, qz, qw]
+        """
+        path_poses = []
+        for tf in path_transformations:
+            (x, y, z) = tf[0:3, -1]
+            (Qxx, Qyy, Qzz, _) = np.diag(tf)
+            t = np.trace(tf[:-1, :-1])
+            r = np.sqrt(1+t)
+
+            # TODO: Fix (invalid value warning) so copysign quant calculation handles imaginary values (or only uses real?)
+            #  assigned to @ACBuynak
+            qx = np.copysign(np.absolute(0.5*np.lib.scimath.sqrt(1 + Qxx - Qyy - Qzz)), tf[2, 1]-tf[1, 2])
+            qy = np.copysign(np.absolute(0.5*np.lib.scimath.sqrt(1 - Qxx + Qyy - Qzz)), tf[0, 2]-tf[2, 0])
+            qz = np.copysign(np.absolute(0.5*np.lib.scimath.sqrt(1 - Qxx - Qyy + Qzz)), tf[1, 0]-tf[0, 1])
+            qw = 0.5*r
+
+            qx = 0 if np.isnan(qx) else qx
+            qy = 0 if np.isnan(qy) else qy
+            qz = 0 if np.isnan(qz) else qz
+
+            path_poses.append([x, y, z, qx, qy, qz, qw])
+
+        return path_poses
+
+
+# todo Remove or refactor this code. For @ACBuynak
 def main():
-  print("-- DEMO ONLY --\nACTUAL TOOLS STORED IN 'transformations' CLASS\n")
-
-  ## Imports
-  import numpy as np
+  print "-- DEMO ONLY --\nACTUAL TOOLS STORED IN 'transformations' CLASS\n"
 
-  ## Instantiate Class
-  tf = transformations()
+  tf = Transformations()
```

Comment:
Just pointing this out here as an example.. 
Throughout this commit you overwrite code that already exists with the exact same thing, these sort of non-changes should be avoided as it makes it more difficult to trace through the history of the code and see what was worked on or where new bugs could have been introduced.

Ideally, you can avoid this through just using a git client to build your commit and include/exclude the real changes you want to push.

Reply:
Are you trying to push GitKraken again? 😛  Maybe I should start using that. Maybe it'd help me understand **just** how much I've messed up my commits haha

Reply:
Maybe. Depending on how you've been using PyCharm, you might be getting the same benefit already.